### DISCUSSION TITLE
`KP_` (numeric keypad) keys not working?

### DISCUSSION THREAD

**Discussion by xfzv** [wezterm/wezterm#3659] [Q&A] [answered]
```console
❯ wezterm --version
wezterm 20230408-112425-69ae8472-gentoo
```

Here's my config:

```lua
-- $HOME/.config/wezterm/wezterm.lua
config.keys = require("keys")
-- [..]

-- $HOME/.config/wezterm/keys.lua
local wt = require("wezterm")
local act = wt.action
local keys = {}

local mapkey = function(mods, key, action)
    keys[#keys + 1] = { key = key, mods = mods, action = action }
end

mapkey("CTRL|SHIFT", "Add", act.IncreaseFontSize)
mapkey("CTRL|SHIFT", "Subtract", act.DecreaseFontSize)

return keys
```

I'd like to assign `KP_Add` and `KP_Subtract` to increase/decrease font size but it doesn't work. I'm not getting any error after saving the changes in `wezterm.lua` but when pressing the keys combination, "+" and "-" respectively are inserted in the terminal instead of performing the assigned actions.

My other keybindings are working just fine. No issue either if I use other keys, for example:

```lua
mapkey("CTRL|SHIFT", "UpArrow", act.IncreaseFontSize)
mapkey("CTRL|SHIFT", "DownArrow", act.DecreaseFontSize)
```

`KP_Add` and `KP_Subtract` are recognized (and work just fine in Kitty by the way):

```console
❯ xev | awk -F'[ )]+' '/^KeyPress/ { a[NR+2] } NR in a { printf "%-3s %s\n", $5, $8 }'
86  KP_Add
82  KP_Subtract
```

[Configuring Key Assignments](https://wezfurlong.org/wezterm/config/keys.html#configuring-key-assignments) does mention `Add` and `Subtract` and it looks like pretty much all numeric keypad keys are supposed to work I guess?

https://github.com/wez/wezterm/blob/9bc7de70eafcafca6e6e5cf21879f6e964c21448/window/src/os/xkeysyms.rs#L95-L97

Any idea?

Thanks.

**Comment by wez** [comment]
Please try the nightly build, as there were some changes around the numeric keypad recently

**Reply by xfzv** [reply]
Thanks for the suggestion. I've just tried with the nightly AppImage, same issue.

**Reply by wez** [reply, marked as answer]
Please run through https://wezfurlong.org/wezterm/troubleshooting.html#debugging-keyboard-related-issues and then file an issue using the bug report template

**Reply by xfzv** [reply]
Will do. Thank you!
